backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt

```diff
@@ -181,6 +191,20 @@ internal val FunctionDescriptor.functionName: String
 
                     append("objc:")
                     append(it.selector)
+                    if (this@with is IrConstructor && this@with.isObjCConstructor()) append("#Constructor")
+
+                    // We happen to have the clashing combinations such as
+                    //@ObjCMethod("issueChallengeToPlayers:message:", "objcKniBridge1165")
+                    //external fun GKScore.issueChallengeToPlayers(playerIDs: List<*>?, message: String?): Unit
+                    //@ObjCMethod("issueChallengeToPlayers:message:", "objcKniBridge1172")
+                    //external fun GKScore.issueChallengeToPlayers(playerIDs: List<*>?, message: String?): Unit
+                    // So disambiguate by the name of the bridge for now.
+                    // TODO: idealy we'd never generate such identical declarations.
+
+                    if (this@with is IrSimpleFunction && this@with.hasObjCMethodAnnotation()) {
+                        this@with.objCMethodArgValue("selector") ?.let { append("#$it") }
+                        this@with.objCMethodArgValue("bridge") ?.let { append("#$it") }
```

Comment:
`bridge` is not considered public and thus is not guaranteed to be stable. So this approach prevents libraries compiled with different K/N versions from being binary compatible.

Reply:
Any alternative approaches?

Reply:
Considering how important binary compatibility is, there are a lot of alternative approaches worth implementing.

For example, we can filter duplicates out in `cinterop` or handle them properly in the compiler.

Reply:
Btw, `cinterop` is filtering out duplicates in master branch now, so this workaround will likely become redundant after rebase.